crates/circuit/src/dag_circuit.rs

```diff
@@ -3571,6 +3610,35 @@ impl DAGCircuit {
         }
     }
 
+    /// Returns an iterator of the ancestors indices of a node.
+    pub fn ancestors<'a>(&'a self, node: &DAGNode) -> impl Iterator<Item = NodeIndex> + 'a {
+        if let Some(node_index) = node.node {
```

Comment:
I would just do `node.node.unwrap()` for now. Eventually, I will do a pass over everything in our joint PR to take care of a proper check using `#[pyo3(from_py_with = "...")]` (https://pyo3.rs/main/function#per-argument-options).